### PR TITLE
fix for PHP 7.2 API changes

### PR DESCRIPTION
**Pull request by remicollet**
```
PHP_VERSION : 7.2.0alpha2
ZEND_VERSION: 3.2.0-dev
...
Number of tests :   92                92
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   92 (100.0%) (100.0%)

```

### CODE REVIEW

File: php_ref.h [hunk 1]
```diff
@@ -56,4 +56,20 @@ ZEND_EXTERN_MODULE_GLOBALS(ref);
...
+#define REF_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX   ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX
+#define REF_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX    ZEND_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX
+
+#else
+
+#define REF_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(name, return_reference, required_num_args, type,       allow_null) \
+       ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(name, return_reference, required_num_args, type, NULL, allow_null)
+
+#define REF_BEGIN_ARG_WITH_RETURN_OBJ_INFO_EX(name, return_reference, required_num_args,            classname, allow_null) \
+       ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX(name, return_reference, required_num_args, IS_OBJECT, classname, allow_null)
```
Comment (pinepain) [inline php_ref.h:70]: While it's not used in my code, should it be `#classname` and not `classname`?

File: php_ref.h [hunk 2]
```diff
@@ -56,4 +56,20 @@ ZEND_EXTERN_MODULE_GLOBALS(ref);
 ZEND_TSRMLS_CACHE_EXTERN();
 #endif
 
+#if PHP_VERSION_ID >= 70200
+
+#define REF_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX   ZEND_BEGIN_ARG_WITH_RETURN_TYPE_INFO_EX
```
Comment (pinepain) [inline php_ref.h:61]: I normally stick to prefix with full ext name - `PHP_REF`, do you mind to alter?

**Comment by pinepain** [comment]
Thanks for the fix! As to appveyor red build, I just merged a fix to master.

File: php_ref.h (hunk 1 above)
Comment (remicollet) [reply php_ref.h:70]: No, classname is correct, also used in https://github.com/pmjones/ext-request/commit/acaa9671bf73c609f65f42c12c50780d490d5e08 and works (both macro are used there)

**Comment by remicollet** [comment]
ok, so can close this one

File: php_ref.h (hunk 1 above)
Comment (pinepain) [reply php_ref.h:70]: Ah, see that. I just stringify class name as it done in 7.2, but it doesn't actually matter for old code what approach to use. Thanks for the hint!

**Comment by pinepain** [comment]
Merged, thanks!

**Pull request closed without merging**